FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/AzureStorageJobFactory.cs

```diff
@@ -0,0 +1,152 @@
+            _groupMemberExtractor = EnsureArg.IsNotNull(groupMemberExtractor, nameof(groupMemberExtractor));
+            _parquetDataProcessor = EnsureArg.IsNotNull(parquetDataProcessor, nameof(parquetDataProcessor));
+            _fhirSchemaManager = EnsureArg.IsNotNull(fhirSchemaManager, nameof(fhirSchemaManager));
+            _filterManager = EnsureArg.IsNotNull(filterManager, nameof(filterManager));
+            _metadataStore = EnsureArg.IsNotNull(metadataStore, nameof(metadataStore));
+
+            EnsureArg.IsNotNull(schedulerConfiguration, nameof(schedulerConfiguration));
+            _schedulerConfiguration = schedulerConfiguration.Value;
+
+            _loggerFactory = EnsureArg.IsNotNull(loggerFactory, nameof(loggerFactory));
+            _logger = _loggerFactory.CreateLogger<AzureStorageJobFactory>();
+        }
+
+        public IJob Create(JobInfo jobInfo)
+        {
+            EnsureArg.IsNotNull(jobInfo, nameof(jobInfo));
+
+            var taskFactoryFuncs =
+                new Func<JobInfo, IJob>[] { CreateProcessingTask, CreateOrchestratorTask };
+
+            foreach (var factoryFunc in taskFactoryFuncs)
+            {
+                var job = factoryFunc(jobInfo);
+                if (job != null)
+                {
+                    return job;
+                }
+            }
+
+            // job hosting didn't catch any exception thrown during creating job,
+            // return null for failure case, and job hosting will skip it.
+            _logger.LogWarning($"Failed to create job, unknown job definition. ID: {jobInfo?.Id ?? -1}");
+            return null;
+        }
+
+        private IJob CreateOrchestratorTask(JobInfo jobInfo)
+        {
+            try
+            {
+                var inputData = JsonConvert.DeserializeObject<FhirToDataLakeOrchestratorJobInputData>(jobInfo.Definition);
+                if (inputData is { JobType: JobType.Orchestrator })
+                {
+                    var currentResult = string.IsNullOrWhiteSpace(jobInfo.Result)
+                        ? new FhirToDataLakeOrchestratorJobResult()
+                        : JsonConvert.DeserializeObject<FhirToDataLakeOrchestratorJobResult>(jobInfo.Result);
+
+                    return new FhirToDataLakeOrchestratorJob(
+                        jobInfo,
+                        inputData,
+                        currentResult,
+                        _dataClient,
+                        _dataWriter,
+                        _queueClient,
+                        _groupMemberExtractor,
+                        _filterManager,
+                        _metadataStore,
+                        _schedulerConfiguration,
+                        _loggerFactory.CreateLogger<FhirToDataLakeOrchestratorJob>());
+                }
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, "Failed to create orchestrator job.");
```

Comment:
Nit: Maybe log warning here? and for line 145

Reply:
fixed.